apps/estuary/test/integration/estuary_test.exs

```diff
@@ -0,0 +1,119 @@
+defmodule Estuary.EstuaryTest do
+  use ExUnit.Case
+  use Placebo
+  use Divo
+  import SmartCity.TestHelper, only: [eventually: 1]
+
+  @elsa_endpoint Application.get_env(:estuary, :elsa_endpoint)
+  @event_stream_topic Application.get_env(:estuary, :event_stream_topic)
+  @event_stream_table_name Application.get_env(:estuary, :event_stream_table_name)
+
+  test "Topic is created when Estuary starts" do
+    assert Elsa.Topic.exists?(@elsa_endpoint, @event_stream_topic)
+  end
+
+  test "should create event_stream table when estuary starts" do
+    expected_table_value = [[@event_stream_table_name]]
+
+    actual_table_value =
+      "SHOW TABLES LIKE '#{@event_stream_table_name}'"
+      |> Prestige.execute()
+      |> Prestige.prefetch()
+
+    assert expected_table_value == actual_table_value
+  end
+
+  test "should check all the columns exists in the table when estuary starts" do
+    expected_column_value = [
+      ["author", "varchar", "", ""],
+      ["create_ts", "bigint", "", ""],
+      ["data", "varchar", "", ""],
+      ["type", "varchar", "", ""]
+    ]
+
+    actual_column_value =
+      "DESCRIBE #{@event_stream_table_name}"
+      |> Prestige.execute()
+      |> Prestige.prefetch()
+
+    assert expected_column_value == actual_column_value
+  end
+
+  test "estuary persists event to the event_stream table" do
+    produce_event(
+      @event_stream_topic,
+      ~s({"__brook_struct__":"Elixir.Brook.Event","__struct__":"Elixir.SmartCity.Dataset","author":"reaper","create_ts":5,"data":"some data","forwarded":false,"type":"some type"})
+    )
+
+    eventually(fn ->
+      events =
+        Prestige.execute(
+          "SELECT COUNT(*) from #{@event_stream_table_name} WHERE author = 'reaper' and create_ts = 5 and data = 'some data' and type = 'some type'"
+        )
+        |> Prestige.prefetch()
+
+      assert events == [[1]]
+    end)
+
+    Prestige.execute("DELETE from #{@event_stream_table_name}") |> Prestige.prefetch()
+  end
+
+  test "estuary persists batch of events to the event stream" do
+    produce_event(@event_stream_topic, [
+      ~s({"__brook_struct__":"Elixir.Brook.Event","__struct__":"Elixir.SmartCity.Dataset","author":"forklift","create_ts":1,"data":"some data1","forwarded":false,"type":"some type1"}),
+      ~s({"__brook_struct__":"Elixir.Brook.Event","__struct__":"Elixir.SmartCity.Dataset","author":"valkyrie","create_ts":2,"data":"some data2","forwarded":false,"type":"some type2"})
+    ])
+
+    eventually(fn ->
+      events =
+        Prestige.execute(
+          "SELECT COUNT(*) from #{@event_stream_table_name} WHERE author = 'forklift' and create_ts = 1 and data = 'some data1' and type = 'some type1'"
+        )
+        |> Prestige.prefetch()
+
+      assert events == [[1]]
```

Comment:
I think `assert events == [[1]]` works well, but I'd like it even more if we did a select and saw we were getting an actual record back. That would prove that we don't always get [[1]] back for any query, it be different between tests, and it would explain what we expect to get back from prestige.